core/src/main/java/org/jruby/RubyHash.java

```diff
@@ -1615,21 +1640,48 @@ public IRubyObject transform_values(final ThreadContext context, final Block blo
         return hashCopy(context).transform_values_bang(context, block);
     }
 
-    @JRubyMethod(name = "transform_keys!")
-    public IRubyObject transform_keys_bang(final ThreadContext context, final Block block) {
+    @JRubyMethod(name = "transform_keys!", optional =  1, rest = true)
+    public IRubyObject transform_keys_bang(final ThreadContext context, IRubyObject[] arg0, final Block block) {
+        RubyArray transformKeys = null;
+
+        if((arg0.length > 0) && (arg0 [0] != null))
```

Comment:
No space between 'arg0 [0]'.  Make it arg0[0].  We also name list of arguments args by convention.